password_enhancements.module

```diff
@@ -0,0 +1,471 @@
+<?php
+
+/**
+ * @file
+ * Password related enchantments.
+ */
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\password_enhancements\ConfigTranslation\ConstraintMapper;
+use Drupal\password_enhancements\Entity\Constraint;
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_theme().
+ */
+function password_enhancements_theme($existing, $type, $theme, $path): array {
+  return [
+    'password_enhancements_error_messages' => [
+      'variables' => [
+        'attributes' => NULL,
+        'messages' => [],
+        'title' => '',
+        'title_attributes' => NULL,
+      ],
+    ],
+    'password_enhancements_policy_constraint' => [
+      'variables' => [
+        'description' => '',
+        'attributes' => NULL,
+      ],
+    ],
+    'password_enhancements_policy_constraints' => [
+      'variables' => [
+        'attributes' => NULL,
+        'constraints' => [],
+        'descriptions' => [],
+        'title' => '',
+        'title_attributes' => [],
+      ],
+    ],
+    'password_enhancements_logged_out_message' => [
+      'variables' => [
+        'attributes' => NULL,
+        'message' => '',
+      ],
+    ],
+  ];
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function password_enhancements_form_user_form_alter(&$form, FormStateInterface $form_state, string $form_id): void {
+  _password_enhancements_add_password_requirement_validation($form, $form_state);
+  $form['password_enhancements_password_change_required']['#access'] = Drupal::currentUser()->hasPermission('administer user password settings');
```

Comment:
This should be handled by field access instead because this field value should not be changed by non-evaluated users wherever this field gets exposed by a developer (accidentally).